File: Код/main5.py
from math import log10


def main():
    for num in range(10, 10000):
        length = int(log10(num)) + 1
        ams = 0

        for p in range(length):
            ams += (int(num / 10 ** p) % 10) ** length

        if num == ams:
            print(num, end=", ")


if __name__ == '__main__':
    main()

File: Код/Main5.java
public class Main5 {

    public static void main(String[] args) {
        for (int num = 10; num < 10000; num++) {
            int len = (int) Math.log10(num) + 1;
            int ams = 0;

            for (int div = 1; div <= num; div *= 10) {
                ams += Math.pow((float) ((num / div) % 10), len);
            }

            if (num == ams) {
                System.out.print(num + ", ");
            }
        }
    }
}
